### PR TITLE
General development dir translation and include img

### PR DESCRIPTION
**Pull request by imlidian**

### CODE REVIEW

File: java-chassis-reference/en_US/general-development/dnsconfig.md [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
-| 项目 | tag |
+| Project | tag |
 | :--- | :--- |
-| 服务中心 | sc.consumer |
-| 配置中心 | cc.consumer |
-| 看板中心 | mc.consumer |
-| 用户自定义 | self.tag |
+| Service Center | sc.consumer |
+| Configuration Center | cc.consumer |
+| Kanban Center | mc.consumer |
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/dnsconfig.md]: 没有看板

File: java-chassis-reference/en_US/general-development/AlarmEvent.md [hunk 2]
```diff
@@ -1,11 +1,11 @@
-# 获取熔断与实例隔离告警事件
+# Get the fuse and instance isolation alarm events
```
[the remark this answers is not in the document]
Comment (imlidian) [reply java-chassis-reference/en_US/general-development/AlarmEvent.md]: Get warning event from Circuit Breaking and Instance Isolation.

File: java-chassis-reference/en_US/general-development/AlarmEvent.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-## 场景描述
-* 在微服务运行期间熔断或实例隔离状态发生变化时，需要监听到相关事件，获取相关信息并进行处理
+## Scene Description
+* When the microservice is running or the instance isolation status changes, you need to listen to related events, get relevant information and process it.
 
-## 使用参考
+## Use Reference
 
-* 监听熔断事件
+* Monitor blown events
```
[the remark this answers is not in the document]
Comment (imlidian) [reply java-chassis-reference/en_US/general-development/AlarmEvent.md]: CircuitBreaker全球统一

File: java-chassis-reference/en_US/general-development/cross-app-invocation.md [hunk 4]
```diff
@@ -33,14 +33,14 @@ consumer端指定微服务名称调用provider的时候，需要加上provider
...
+- RPC call method
 
-  当consumer端以RPC方式开发微服务消费者时，声明的服务提供者代理如下：
+  When the consumer client develops a microservice consumer in RPC mode, the declared service provider proxy is as follows:
   ```java
     @RpcReference(schemaId = "hello", microserviceName = "helloApp:helloProvider")
     private Hello hello;
   ```
-  调用方式和调用同应用下的微服务相同：
+  The calling method and calling are the same as the microservices under the application:
```
[the remark this answers is not in the document]
Comment (imlidian) [reply java-chassis-reference/en_US/general-development/cross-app-invocation.md]: 中文就诡异

File: java-chassis-reference/en_US/general-development/dai-li-she-zhi.md [hunk 5]
```diff
@@ -1,32 +1,29 @@
...
-    passwd: yourpassword    #密码
+  proxy:
+    enable: true #Do you want to enable the proxy?
+    host: yourproxyaddress #proxy address
+    port: 80 #proxy port
+    username: yourname #username
+    passwd: yourpassword #password
 ```
 
-**注意：当前仅支持连接服务中心、配置中心、服务看板支持代理，如果对接其他三方服务，可以读取这个配置，自行配置代理，vertx httpclient支持代理设置，例如：**
```
[the remark this answers is not in the document]
Comment (imlidian) [reply java-chassis-reference/en_US/general-development/dai-li-she-zhi.md]: 啥意思 不支持这个？

File: java-chassis-reference/en_US/general-development/reactive.md [hunk 6]
```diff
@@ -34,45 +34,45 @@ public String hello(@PathVariable(name = "name") String name){
...
+* Because it is a synchronous call, the calling thread of "Microservice A" is always in the blocking wait state before "Other microservices" is not answered, and does not process any other transactions.
 
-* 当Executor中所有线程都在等待远程应答时，所有新的请求都只能在Queue中排队，得不到处理，此时整个系统相当于停止工作了
+* When all threads in the Executor are waiting for a remote response, all new requests can only be queued in the Queue and cannot be processed. At this point, the entire system is equivalent to stop working.
 
-* 要增加处理能力，只能增加Executor中的线程数，而操作系统并不能无限地增加线程数，事实上增加线程数带来的收益是一个抛物线模型，超出一定的临界值后，系统的处理能力其实会下降，而这个临界值并不会太大
+* To increase the processing power, only increase the number of threads in the Executor, and the operating system can not increase the number of threads indefinitely. The benefit of increasing the number of threads is a parabolic model. After a specific critical value, the system handles The ability will drop, and this threshold will not be too big.
 
-* 当业务逻辑中，需要多次进行远程同步操作时，会更加恶化这个现象
+* This phenomenon is exacerbated when the remote synchronization operation is required multiple times in the business logic.
```
[the remark this answers is not in the document]
Comment (imlidian) [reply java-chassis-reference/en_US/general-development/reactive.md]: exacerbated the problem

**Comment by imlidian** [comment]
all done

File: java-chassis-reference/en_US/general-development/file-upload.md [hunk 7]
```diff
@@ -1,54 +1,54 @@
...
 ```java
 @PostMapping(path = "/upload", consumes = MediaType.MULTIPART_FORM_DATA)
 public String fileUpload(@RequestPart(name = "file1") MultipartFile file1, @RequestPart(name = "file2") Part file2, @RequestPart String param1) {
-  ……
+  ......
 }
 ```
 
-### 配置说明：
+### Configuration instructions:
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/file-upload.md]: instructions -> Introduction

File: java-chassis-reference/en_US/general-development/file-upload.md [hunk 8]
```diff
@@ -1,54 +1,54 @@
...
 
-| 配置项 | 默认值 | 取值范围 | 含义 |
+| Configuration Item | Default Value | Range of Value |
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/file-upload.md]: 含义列没有了？？？

File: java-chassis-reference/en_US/general-development/file-upload.md [hunk 9]
```diff
@@ -58,25 +58,25 @@ public String fileUpload(@RequestPart(name = "file1") MultipartFile file1, @Requ
 
 * org.springframework.core.io.Resource
 
-使用InputStream时，因为是流的方式，此时没有客户端文件名的概念，所以producer获取客户端文件名会得到null
+When using InputStream, because it is a stream, there is no concept of client file name at this time, so the producer will get the client file name will get null.
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/file-upload.md]: so the producer will get the client file name will get null ->
so the producer gets the client file name will be null

File: java-chassis-reference/en_US/general-development/file-upload.md [hunk 10]
```diff
@@ -58,25 +58,25 @@ public String fileUpload(@RequestPart(name = "file1") MultipartFile file1, @Requ
...
 
-如果既要使用内存数据，又想让producer可以获取客户端文件名，可以使用resource类型，继承org.springframework.core.io.ByteArrayResource，且override getFilename即可。
+If you want to use both memory data and the producer to get the client file name, you can use the resource type, inherit org.springframework.core.io.ByteArrayResource, and override getFilename.
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/file-upload.md]: If you want to use both memory data and the producer to get the client file name ->
If you want to use memory data and the producer also can get the client file name

File: java-chassis-reference/en_US/general-development/local-develop-test.md [hunk 11]
```diff
@@ -24,65 +24,68 @@
...
+Servicecomb:
+  Service:
+    Registry:
+      Address:
+        #Service Center address and port
+        Http://127.0.0.1:30100
 ```
 
-* **步骤 3 **开发服务提供/消费者，启动微服务进行本地测试。
+* **Step 3 **Development service provider/consumer, launch microservices for local testing.
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/local-develop-test.md]: service provider/consumer -> microservices

File: java-chassis-reference/en_US/general-development/local-develop-test.md [hunk 12]
```diff
@@ -24,65 +24,68 @@
...
 
-**----结束**
+**----End**
 
-## Mock机制启动服务中心
-在本进程内存中模拟一个只能本进程使用的服务中心，一般是在测试场景中使用。
-* ### 进程内调用
-只需要在启动ServiceComb引擎之前声明一下即可启用：
+## Mock mechanism start service center
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/local-develop-test.md]: Mock mechanism start service center ->
Using mock mechanism to start ServiceCenter

File: java-chassis-reference/en_US/general-development/local-develop-test.md [hunk 13]
```diff
@@ -24,65 +24,68 @@
...
 System.setProperty("local.registry.file", "notExistJustForceLocal");
 ```
-* ### 跨进程调用
-如果部署比较简单，并且部署信息是静态的，即使有跨进程调用也可以使用本Mock机制
-producer端仍然像“进程内调用”一样声明即可
-但是，因为Mock并不能跨进程生效，所以consumer端的Mock，需要提供一个本地的配置文件，在里面描述调用目标的详细信息，包括名字、版本、地址、schema id等等信息
-同样，因为Mock不能跨进程，consumer也无法动态取得producer的契约信息，所以，需要在本地提供契约文件
-（这个场景，使用Mock服务中心，比使用standalone的服务中心，成本高得多得多，不建议使用）
+* ### Cross-process call
+If the deployment is simple and the deployment information is static, you can use this mock mechanism even if you have a cross-process call.
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/local-develop-test.md]: cross-process call ->
remote process call

File: java-chassis-reference/en_US/general-development/local-develop-test.md [hunk 14]
```diff
@@ -96,28 +99,32 @@ localserv:
         - rest://localhost:8080
         - highway://localhost:7070
 ```
-* **步骤 2**consumer本地部署契约文件
+* **Step 2**consumer local deployment contract file
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/local-develop-test.md]: Deploy local contract file in consumer

File: java-chassis-reference/en_US/general-development/local-develop-test.md [hunk 15]
```diff
@@ -96,28 +99,32 @@ localserv:
...
 
-参考：[定义服务契约](https://huaweicse.github.io/servicecomb-java-chassis-doc/zh_CN/build-provider/define-contract.html)
-* **步骤 3**在consumer main函数，启动ServiceComb引擎之前声明：
+Reference: [Define Service Contract] (https://docs.servicecomb.io/java-chassis/zh_CN/build-provider/define-contract.html)
+* **Step 3** In the consumer main function, declare the ServiceComb engine before starting:
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/local-develop-test.md]: declare the ServiceComb engine before starting ->
declare it before  the ServiceComb engine starting

File: java-chassis-reference/en_US/general-development/local-develop-test.md [hunk 16]
```diff
@@ -96,28 +99,32 @@ localserv:
...
 　　System.setProperty("local.registry.file", "/path/registry.yaml");
 ```
 
-setProperty第二个参数填写registry.yaml在磁盘中的系统绝对路径，注意区分在不同系统下使用对应的路径分隔符。
+The second parameter of setProperty fills in the absolute path of the registry.yaml system on the disk, pay attention to distinguish the corresponding path separator in different systems.
+
 
+## Local debugging by setting environment information
+The java chassis is strictly dependent on the contract when designing, so usually the version of the microservice has to change when the contract updated. However, if the development mode is still in progress, it is normal to modify the interface.   
+If you need to change the version every time, it is very unfriendly to the user, so an environment setting is added for this case. If the microservice is configured as a development environment, the interface is modified (the schema has changed), and the restart can be registered to the service center without modifying the version number.
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/local-develop-test.md]: If you need to change the version every time ->
If need to change the version every time

File: java-chassis-reference/en_US/general-development/CORS.md [hunk 17]
```diff
@@ -1,29 +1,29 @@
...
+When the user needs to send REST requests from across the browser, the CORS mechanism may be used. The microservices that receive cross-domain requests need to enable CORS support.
 
-## 配置说明
+## Configuration instructions
 
-CORS功能在microservice.yaml文件中配置，配置项见下表所述。
+The CORS function is configured in the microservice.yaml file. The configuration items are described in the following table.
 
-| 配置项 | 默认值 | 取值范围 | 是否必选 | 含义 | 注意 |
+| Configuration Item | Default Value | Range of Value | Required | Meaning |
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/CORS.md]: 少了一个 “注意” 列

File: java-chassis-reference/en_US/general-development/CORS.md [hunk 18]
```diff
@@ -1,29 +1,29 @@
...
-| servicecomb.cors.maxAge | 无 | (0,2147483647]，整型 | 否 | Access-Control-Max-Age | 单位是秒，如果用户不配置此项，则CORS应答中没有Access-Control-Max-Age |
+| servicecomb.cors.enabled | `false` | `true`/`false` | No | Whether to enable CORS function | - |
+| servicecomb.cors.origin | `*` | - | No | Access-Control-Allow-Origin | - |
+| servicecomb.cors.allowCredentials | `false` | `true`/`false` | No | Access-Control-Allow-Credentials | According to the CORS standard, when Access-Control-Allow-Credentials is set to `true`, Access- Control-Allow-Origin cannot be set to "*", otherwise an exception will be thrown |
+| servicecomb.cors.allowedHeader | None | - | No | Access-Control-Allow-Headers | Multiple values ​​separated by commas |
+| servicecomb.cors.allowedMethod | None | - | No | Access-Control-Allow-Methods | Multiple values ​​separated by commas |
+| servicecomb.cors.exposedHeader | None | - | No | Access-Control-Expose-Headers | Multiple values ​​separated by commas |
+| servicecomb.cors.maxAge | None | (0,2147483647], Integer | No | Access-Control-Max-Age | The unit is seconds. If the user does not configure this, there is no Access-Control-Max in the CORS response. Age |
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/CORS.md]: The unit is seconds. If the user does not configure this, there is no Access-Control-Max in the CORS response. Age
The unit is seconds. If the user does not configure this, there is no Access-Control-Max-Age in the CORS response.

File: java-chassis-reference/en_US/general-development/context.md [hunk 19]
```diff
@@ -1,36 +1,36 @@
-# 使用Context传递控制消息
+# Delivery Context Control Message
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/context.md]: Delivery Control Message via Context

File: java-chassis-reference/en_US/general-development/cross-app-invocation.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (zhengyangyong) [reply java-chassis-reference/en_US/general-development/cross-app-invocation.md]: The calling method and calling are the same as the microservices under the application: -> 
The calling method is same as calling microservices under the same application 
这样是不是更好一点

File: java-chassis-reference/en_US/general-development/dai-li-she-zhi.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (zhengyangyong) [reply java-chassis-reference/en_US/general-development/dai-li-she-zhi.md]: 是的，开源版本ServiceComb没有看板服务

File: java-chassis-reference/en_US/general-development/dnsconfig.md [hunk 20]
```diff
@@ -1,41 +1,41 @@
-## 场景描述
+## Scene Description
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/dnsconfig.md]: Description -> Introduction

File: java-chassis-reference/en_US/general-development/dnsconfig.md (hunk 1 above)
Comment (zhengyangyong) [reply java-chassis-reference/en_US/general-development/dnsconfig.md]: 移除Kanban Center这一行

File: java-chassis-reference/en_US/general-development/dnsconfig.md [hunk 21]
```diff
@@ -1,41 +1,41 @@
...
+| Kanban Center | mc.consumer |
+| User Defined | self.tag |
 
-各个properties详细说明（设置Vertx DNS解析）    
+The detailed description of each property (Set Vertx DNS resolution)
 
 ``` yaml
 addressResolver:
-  servers: 8.8.8.8,8.8.4.4   #对应Linux /etc/resolv.conf的nameserver,DNS服务器地址，支持配置多个，以逗号隔开
-  ndots: 1                   #对应linux /etc/resolv.conf里面的options: ndots, 作用就是如果给的域名里面包含的点的个数少于该阈值，那么DNS解析的时候就会默认加上searchDomains的值，这个必须和searchDomains搭配使用，Linux默认为1，华为公有云PAAS（包含容器）默认是4
```
[the remark this answers is not in the document]
Comment (zhengyangyong) [reply java-chassis-reference/en_US/general-development/dnsconfig.md]: 删掉“华为公有云PAAS（包含容器）默认是4”

File: java-chassis-reference/en_US/general-development/dnsconfig.md [hunk 22]
```diff
@@ -1,41 +1,41 @@
...
-  optResourceEnabled: true   #optional record is automatically included in DNS queries
-  cacheMinTimeToLive: 0      #最小缓存时间
-  cacheMaxTimeToLive: 10000  #最大缓存时间
-  cacheNegativeTimeToLive: 0 #DNS解析失败后，下次重试的等待时间
-  queryTimeout: 5000         #查询超时时间
-  maxQueries: 4              #查询次数
-  rdFlag: true               #设置DNS递归查询
-  rotateServers: true        #设置是否支持轮询
+  servers: 8.8.8.8, 8.8.4.4 #corresponds to the nameserver of Linux /etc/resolv.conf, the DNS server address, supports multiple configurations, separated by commas
+  ndots: 1 # corresponds to the options in linux /etc/resolv.conf: ndots, the role is that if the number of points contained in the domain name is less than the threshold, then DNS resolution will be added by default to the value of searchDomains. This must be used in conjunction with searchDomains. Linux defaults to 1. Huawei public cloud PAAS (including containers) defaults to 4
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/dnsconfig.md]: 删掉Huawei public cloud PAAS (including containers) defaults to 4

File: java-chassis-reference/en_US/general-development/produceprocess.md [hunk 23]
```diff
@@ -1,23 +1,23 @@
-# 返回值序列化扩展
-## 概念阐述
+# return value serialization extension
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/produceprocess.md]: 第一个字母R大写

File: java-chassis-reference/en_US/general-development/produceprocess.md [hunk 24]
```diff
@@ -1,23 +1,23 @@
...
+## Concept Description
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/produceprocess.md]: Description -> Introduction

File: java-chassis-reference/en_US/general-development/produceprocess.md [hunk 25]
```diff
@@ -1,23 +1,23 @@
...
+The current REST channel return value supports both application/json and text/plain formats, supports developer extensions and rewrites, service providers provide serialization capabilities through producer declarations, and service consumers specify return value serialization through the request's Accept header. By default, the data in application/json format is returned.
 
-## 开发说明
+## Development Instructions
 
-* ### 扩展
+* ### extension
 
-  开发人员可以根据业务需要，通过编程的方式来扩展返回值序列化方式。实施步骤如下，以扩展支持application/xml格式为例：
+  Developers can extend the return value serialization method programmatically based on business needs. The implementation steps are as follows, taking the extended support application/xml format as an example:
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/produceprocess.md]: the extended -> how to make extension in order to

File: java-chassis-reference/en_US/general-development/secret-field.md [hunk 26]
```diff
@@ -1,22 +1,22 @@
-## 场景描述
+## Scene Description
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/secret-field.md]: Description -> Introduction

File: java-chassis-reference/en_US/general-development/secret-field.md [hunk 27]
```diff
@@ -1,22 +1,22 @@
...
 
-由于HTTP协议的非安全性，在网络中传输的数据能轻易被各种抓包工具监听。在实际应用中，业务对应用或服务间传输的敏感数据有较高的安全要求，这类数据需要特别的加密保护（业务不同对算法要求不同），这样即使内容被截获，也可以保护其中的敏感数据不被轻易获取。
+Due to the non-security of the HTTP protocol, data transmitted over the network can be easily monitored by various packet capture tools. In practical applications, services have high security requirements for sensitive data transmitted between applications or services. Such data requires special encryption protection (different services have different algorithm requirements) so that even if the content is intercepted, it can protect. Sensitive data is not easily obtained.
 
-## 解决方法
+## Solution
 
-服务间的通信离开不序列化和反序列化，对于上述的场景，使用jackson类库提供的 @JsonSerialize 和 @JsonDeserialize 注解功能，对敏感数据定制序列化和反序列化方法，并在定制化的方法中实现加解密功能。
+The communication between services leaves unserialized and deserialized. For the above scenario, the @JsonSerialize and @JsonDeserialize annotation functions provided by the jackson class library are used to customize the serialization and deserialization methods for sensitive data, and in a customized method. Implement encryption and decryption functions.
```
Comment (zhengyangyong) [inline java-chassis-reference/en_US/general-development/secret-field.md]: leaves unserialized and deserialized ->
need serialized and deserialized

**Pull request merged**